pynbody/snapshot/tipsy.py

```diff
@@ -1203,8 +1216,19 @@ def __init__(self, filename, sort=True, paramfile=None):
                 # All star iorders are greater than any gas iorder
                 # so this indicates a bad format. (N.B. there is the
                 # possibility of a false negative)
-                if(testread['iord'][0] < testread['iorderGas'][0]): 
-                    file_structure = np.dtype({'names': ("iord", "iorderGas",
+                if(testread['iord'][0] < testread['iorderGas'][0]):
+                    if tcool == True:
+                        file_structure = np.dtype({'names': ("iord", "iorderGas", "tform",
+                                                             "x", "y", "z",
+                                                             "vx", "vy", "vz",
+                                                             "massform", "rhoform", "tempform", "h2form"),
+                                                   'formats': ('i4', 'i4', 'f8',
```

Comment:
This branch failed on a read of 64-bit iords for me, and I think it's because this is supposed to introduce 64-bit iords here, and uses 32-bit by mistake. Making the change from `i4` to `i8` fixed the problem for me locally.

If `tcool == True`, that means `iSize` is 104 (4\*2 + 8\*12) above, but here you've defined `file_structure` with a size of 96 (2\*4 + 8\*11). Therefore, here we should have 64-bit iords. This is consistent with the later logger info as well.

Reply:
That update seems to work. 

I thought tcoolform was phased out, and didn't realize there could be _both_ 64-bit iords and tcoolform. I think this effectively solves the problem (though again, I think using the metadata in the log file is ultimately the best solution going forward).

Reply:
it is now, but might not be in your runs. This will hopefully ensure backwards compatibility. The problem is that due to the degeneracy, you may have a mismatch in manes (i.e. tcoolform may be marked as h2form). I also agree, I think this pull request will be unnecessary once we get a good metadata system going (assuming we can convert pre-existing files too)